sky/backends/cloud_vm_ray_backend.py

```diff
@@ -265,6 +265,25 @@ def add_prologue(self, job_id: int, is_local: bool = False) -> None:
                 log_to_driver=True,
                 **kwargs
             )
+            def wait_or_fail(futures) -> List[int]:
+                \"\"\"Wait for tasks, if any fails, cancel all unready.\"\"\"
+                returncodes = []
+                # Wait for at leat 1 task to be ready.
+                ready, unready = ray.wait(futures)
+                while unready:
+                    returncodes.extend(ray.get(ready))
+                    if any(r != 0 for r in returncodes):
+                        # ray.cancel without force fails to kill tasks. We use
+                        # force=True to kill unready tasks.
+                        if unready:
+                            for task in unready:
+                                ray.cancel(task, force=True)
+                            # A random number to indicate the worker is forcely
+                            # killed by SkyPilot due to a worker failure.
+                            returncodes.extend([65 for _ in unready])
```

Comment:
Depends on SIGKILL or SIGABRT, we should return 128 + signum: https://chat.openai.com/share/1a5fb45b-2e92-406f-83ec-89b5501be830

Reply:
Sounds good! Changed to 137 instead.